From 420e41f8c6c5d669940be9542ce4058f579288b7 Mon Sep 17 00:00:00 2001
From: Konstantin Porotchkin <kostap@marvell.com>
Date: Thu, 6 Aug 2015 17:21:32 +0300
Subject: [PATCH 156/538] mmc: sdhci-pxav3: Change card detection for
 armada-38x-gp.dts to dat3-cd

- Since Marvell GP board revision ID v1.5, the SD card is detected using
  DAT3 detection method. This patch adapts the matching customer DT
  accordingly.
- Affected configurations - armada-380-gp.dts and armada-388-gp.dts

Signed-off-by: Konstantin Porotchkin <kostap@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/22606
Tested-by: Star_Automation <star@marvell.com>
Reviewed-by: Nadav Haklai <nadavh@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/22662

Conflicts:
	arch/arm/boot/dts/armada-388-gp.dts

Change-Id: I046769a064c98c3c56275472691cc824723eccc7
Reviewed-on: http://vgitil04.il.marvell.com:8080/27706
Reviewed-by: Lior Amsalem <alior@marvell.com>
Tested-by: Lior Amsalem <alior@marvell.com>
---
 arch/arm/boot/dts/armada-380-gp.dts | 2 ++
 arch/arm/boot/dts/armada-388-gp.dts | 3 ++-
 2 files changed, 4 insertions(+), 1 deletion(-)

diff --git a/arch/arm/boot/dts/armada-380-gp.dts b/arch/arm/boot/dts/armada-380-gp.dts
index 0958f88..f74db5f 100644
--- a/arch/arm/boot/dts/armada-380-gp.dts
+++ b/arch/arm/boot/dts/armada-380-gp.dts
@@ -177,6 +177,8 @@
 
 			sdhci@d8000 {
 				no-1-8-v;
+				dat3-cd;
+				cd-inverted;
 				wp-inverted;
 				bus-width = <8>;
 				status = "okay";
diff --git a/arch/arm/boot/dts/armada-388-gp.dts b/arch/arm/boot/dts/armada-388-gp.dts
index 571f6e8..cc3554c 100644
--- a/arch/arm/boot/dts/armada-388-gp.dts
+++ b/arch/arm/boot/dts/armada-388-gp.dts
@@ -229,7 +229,8 @@
 				 * 'dat3-cd;'
 				 * 'cd-inverted;'
 				 */
-				broken-cd;
+				dat3-cd;
+				cd-inverted;
 				wp-inverted;
 				bus-width = <8>;
 				status = "okay";
-- 
1.9.1

